src/model/transaction/randomizeBlockMapKeys.js

```diff
@@ -23,7 +23,11 @@ const generateRandomKey = require('generateRandomKey');
 const {OrderedMap} = Immutable;
 
 const randomizeContentBlockNodeKeys = (blockMap: BlockMap): BlockMap => {
-  const newKeys = [];
+  const newKeysRef = {};
+
+  // we keep track of root blocks in order to update subsequent sibling links
+  let lastRootBlock: ContentBlockNode;
```

Comment:
we need to keep track of previous seen root blocks, since randomizeBlockMapKeys is responsible for dealing with fragments, example is when user select some children of a block for copy and paste without including all siblings and or parent, in that case the fragment need to have the parent links cleaned up

Reply:
This all looks great! Nice refactor, plus fixing the parent links like you mentioned above.